### PR TITLE
Implementation of a filter method for Navier

### PR DESCRIPTION
**Pull request by jandrej**
This PR implements the interpolation based filter method presented in [1].

The idea is to interpolate the velocity to a lower polynomial order space (e.g. remove the highest order mode from the solution). The user can choose the number of modes to cut as well as the contribution of the original solution to the filtered result.

This method helps to preserve stability for highly turbulent cases.

[1] Paul Fischer, Julia Mullen: Filter-based stabilization of spectral element methods

Double shear layer roll-up with 0 (!) kinematic viscosity.

<center>
Screen Shot 2020-10-07 at 4 56 58 PM</center>

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1805](https://github.com/mfem/mfem/pull/1805) | @jandrej | @tzanio | @jamiebramwell + @kmittal2 + @tzanio | 10/10/20 | 10/29/20 | 11/03/20 | |

### CODE REVIEW

File: miniapps/navier/navier_solver.cpp [hunk 1]
```diff
@@ -290,6 +290,21 @@ void NavierSolver::Setup(double dt)
 
    un_gf.GetTrueDofs(un);
 
+   if (filter_alpha != 0.0)
```
Comment (tzanio) [inline miniapps/navier/navier_solver.cpp:293]: Document in `CHANGELOG` and maybe mention the reference in the code?

Comment (jandrej) [reply miniapps/navier/navier_solver.cpp:293]: I mentioned the reference. This is the first release with `Navier` officially included so I don't know if this should be mentioned in the `CHANGELOG`.

File: miniapps/navier/navier_solver.hpp [hunk 2]
```diff
@@ -241,6 +241,19 @@ class NavierSolver
...
+
+   /// Set the interpolation filter parameter @a alpha
+   /**
+    * If @a a is > 0, the filtering algorithm for the velocity field after every
+    * time step from [1] is used.
+    *
+    * [1] Paul Fischer, Julia Mullen (2001) Filter-based stabilization of
+    * spectral element methods
+    */
+   void SetFilterAlpha(double a) { filter_alpha = a; }
```
Comment (kmittal2) [inline miniapps/navier/navier_solver.hpp:255]: Check if `0 >= filter_alpha <= 1` ?

File: miniapps/navier/navier_solver.hpp [hunk 3]
```diff
@@ -241,6 +241,19 @@ class NavierSolver
    /// Compute CFL
    double ComputeCFL(ParGridFunction &u, double dt);
 
+   /// Set the number of modes to cut off in the interpolation filter
+   void SetCutoffModes(int c) { filter_cutoff_modes = c; }
+
+   /// Set the interpolation filter parameter @a alpha
```
Comment (kmittal2) [inline miniapps/navier/navier_solver.hpp:247]: Do you mean `filter_alpha` or `a` here?

**Review by kmittal2 — APPROVED** [review]
Looks good to me.

**Comment by tzanio** [comment]
@jamiebramwell, can you take a look?

File: miniapps/navier/navier_solver.cpp (hunk 1 above)
Comment (tzanio) [reply miniapps/navier/navier_solver.cpp:293]: > This is the first release with Navier officially included so I don't know if this should be mentioned in the CHANGELOG.

It is included in 4.2, where it is mentioned... or do you mean something else?

**Review by tzanio — APPROVED** [review]

**Review by jamiebramwell — APPROVED** [review]
Awesome stuff!

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Comment by v-dobrev** [comment]
Merged in post v4.2 `next`.

File: miniapps/navier/navier_solver.cpp (hunk 1 above)
Comment (tzanio) [reply miniapps/navier/navier_solver.cpp:293]: Do you want to update the `CHANGELOG` now?

**Pull request merged**